src/actions/arcActions.ts

```diff
@@ -612,30 +609,44 @@ export function voteOnProposal(daoAvatarAddress: string, proposalId: string, vot
 
       const voteTransaction = await votingMachineInstance.vote({ proposalId, vote });
     } catch (err) {
-      dispatch(showAlert(('Voting failed: ' + err.message)));
       dispatch({ type: arcConstants.ARC_VOTE_REJECTED, payload: err.message });
     }
   };
 }
 
+export type StakeAction = IAsyncAction<'ARC_STAKE', {
+  avatarAddress: string,
+  proposalId: string,
+  stake: number,
+  prediction: VoteOptions,
+  stakerAddress: string,
+}, {
+  proposal: any,
+  stake: any,
+}>
+
 export function stakeProposal(daoAvatarAddress: string, proposalId: string, prediction: number, stake: number) {
   return async (dispatch: Redux.Dispatch<any>, getState: () => IRootState) => {
     const web3: Web3 = Arc.Utils.getWeb3();
     const currentAccountAddress: string = getState().web3.ethAccountAddress;
-
-    // TODO: num transactions pending...
-    let payload: any = {
-      stake: {
-        avatarAddress: daoAvatarAddress,
-        proposalId,
-        stake,
-        prediction,
-        stakerAddress: currentAccountAddress,
-        transactionState: TransactionStates.Unconfirmed,
-      },
+    const proposal: IProposalState = getState().arc.proposals[proposalId];
+
+    const meta = {
```

Comment:
Do you really need all of these for the meta? seems like proposalId and stakerAddress would be enough to identify, since you shouldn't be able to stake twice at the same time on the same proposal

Reply:
at a minimum I think what characterizes a staking operation uniquely is:
1. `avatarAddress` - aka on which DAO?
2. `proposalId` - aka on which proposal?
3. `stakerAddress` - aka who is staking?

But I thought adding `prediction` and `stake` also makes sense and if we use a different `payload` type for the `pending` action from the `success` action, it breaks type-safty. (although we can probably invent a more complicated `AsyncAction` type that would work)